cmd/instrumentless/instrumentless.go

```diff
@@ -50,15 +48,10 @@ func GetCoupon(event string, bearerToken string) (*Coupon, error) {
 		defer res.Body.Close()
 	}
 
-	body, err := ioutil.ReadAll(res.Body)
-	if err != nil {
-		return nil, err
-	}
-
 	coupon := Coupon{}
-	err = json.Unmarshal(body, &coupon)
+	err = json.NewDecoder(res.Body).Decode(&coupon)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("could not get a coupon: %v", err)
```

Comment:
I think this error would be:

> failed to read and parse the response: %v

and "could not get a coupon" would be at the call site of this fn